packages/xo-web/src/xo-app/new-vm/index.js

```diff
@@ -1017,7 +1021,7 @@ export default class NewVm extends BaseComponent {
                   onChange={this._initTemplate}
                   placeholder={_('newVmSelectTemplate')}
                   predicate={this._getVmPredicate()}
-                  value={template}
+                  value={defined(template, this.props.template)}
                 />
               ) : (
                 <SelectResourceSetsVmTemplate
```

Comment:
`SelectResourceSetsVmTemplate`'s value also needs to get the query template value.